### PR TITLE
feat: support osv schema v1.5.0

### LINKED ISSUE #323: Implement OSV schema v1.5.0
See https://github.com/ossf/osv-schema/releases/tag/v1.5.0

### PR DESCRIPTION
**Pull request by JamieMagee**
https://github.com/ossf/osv-schema/releases/tag/v1.5.0

Closes #323

### CODE REVIEW

**Comment by JamieMagee** [comment]
@Churro I can't request a review via the GitHub UI, but I would appreciate your input.

File: packages/osv-offline-db/src/lib/purl-helper.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ const PURL_ECOSYSTEMS: Record<Ecosystem, string> = {
   npm: 'npm',
   Packagist: 'composer',
   PyPI: 'pypi',
+  Pub: 'pub',
```
Comment (Churro) [inline packages/osv-offline-db/src/lib/purl-helper.ts:13]: Would be great to add a test case for pub too

File: packages/osv-offline-db/src/lib/ecosystem.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export const ecosystems = [
   'npm',
   'NuGet',
   'Packagist',
+  'Pub',
```
Comment (Churro) [inline packages/osv-offline-db/src/lib/ecosystem.ts:10]: Just as a note: will need to extend the mapping in renovate too

**Comment by Churro** [comment]
OSV schema changes LGTM.

Back then, I intentionally didn't add pub as there were only 6 advisories (and still not more) and IIRC none of those had a purl identifier, i.e., they could have never been found anyway. It seems this has changed meanwhile, e.g. GHSA-9324-jv53-9cc8 includes "purl": "pkg:pub/dio"`.

**Review by viceice — APPROVED** [review]
new tests for pub would be nice to have

**Pull request merged**